File: app/src/main/java/ru/radiationx/anilibria/utils/mvp/BasePresenter.java
package ru.radiationx.anilibria.utils.mvp;

import com.arellomobile.mvp.MvpPresenter;
import com.arellomobile.mvp.MvpView;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * Created by radiationx on 05.11.17.
 */


public class BasePresenter<ViewT extends MvpView> extends MvpPresenter<ViewT> {
    private CompositeDisposable disposables = new CompositeDisposable();

    public BasePresenter() {
        super();
    }

    @Override
    public void onDestroy() {
        if (!disposables.isDisposed())
            disposables.dispose();
    }

    protected void addDisposable(Disposable disposable) {
        disposables.add(disposable);
    }
}

File: app/src/main/java/ru/radiationx/anilibria/utils/mvp/BasePresenter.kt
package ru.radiationx.anilibria.utils.mvp

import com.arellomobile.mvp.MvpPresenter
import com.arellomobile.mvp.MvpView
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import ru.terrakok.cicerone.Router

/**
 * Created by radiationx on 05.11.17.
 */


open class BasePresenter<ViewT : MvpView>(private val router: Router) : MvpPresenter<ViewT>() {

    private var compositeDisposable = CompositeDisposable()

    override fun onDestroy() {
        compositeDisposable.dispose()
    }

    fun Disposable.addToDisposable() {
        compositeDisposable.add(this)
    }

    fun onBackPressed() {
        router.exit()
    }
}
